src/java.base/windows/classes/sun/nio/ch/WindowsAsynchronousFileChannelImpl.java

```diff
@@ -444,9 +446,10 @@ public void run() {
             long address;
 
             // Substitute a native buffer if not direct
-            if (dst instanceof DirectBuffer) {
+            if (dst.isDirect()) {
                 buf = dst;
                 address = ((DirectBuffer)dst).address() + pos;
+                IOUtil.acquireScope(dst, true);
```

Comment:
I assume we'll have to change this to use IOUtil.bufferAddress(dst), and after the acquire.